reconcile/utils/ocm.py

```diff
@@ -382,20 +406,37 @@ def _get_create_cluster_spec(cluster_name: str, cluster: OCMSpec) -> dict[str, A
             ocm_spec["nodes"]["autoscale_compute"] = autoscale.dict()
         else:
             ocm_spec["nodes"]["compute"] = cluster.spec.nodes
+
+        if isinstance(cluster.spec, ROSAClusterSpec):
+            rosa_spec: dict[str, Any] = {
+                "properties": {
+                    "rosa_creator_arn": cluster.spec.account.rosa_creator_role_arn
+                },
+                "product": {"id": "rosa"},
+                "ccs": {"enabled": True},
+                "aws": {
+                    "account_id": cluster.spec.account.uid,
+                    "sts": {
+                        "enabled": True,
+                        "auto_mode": True,
+                        "role_arn": cluster.spec.account.rosa_installer_role_arn,
+                        "support_role_arn": cluster.spec.account.rosa_support_role_arn,
+                        "instance_iam_roles": {
+                            "master_role_arn": cluster.spec.account.rosa_controlplane_role_arn,
+                            "worker_role_arn": cluster.spec.account.rosa_worker_role_arn,
+                        },
+                        "operator_role_prefix": f"{cluster_name}-{operator_roles_prefix}",
+                    },
+                },
+            }
+
+        ocm_spec.update(rosa_spec)
         return ocm_spec
 
     @staticmethod
     def _get_update_cluster_spec(update_spec: Mapping[str, Any]) -> dict[str, Any]:
         ocm_spec: dict[str, Any] = {}
 
-        instance_type = update_spec.get("instance_type")
```

Comment:
How is it safe to delete this? I assume ROSA does not support this? Then why was this there in the first place?